### PR TITLE
feat: Added Athena resources: data_catalogs, work_groups, named_queries, prepared_statements

### LINKED ISSUE #788: Support Athena
- [ ] [AWS::Athena::DataCatalog](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-athena-datacatalog.html)
- [ ] [AWS::Athena::NamedQuery](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-athena-namedquery.html)
- [ ] [AWS::Athena::PreparedStatement](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-athena-preparedstatement.html)
- [ ] [AWS::Athena::WorkGroup](https://docs.aws.amazon.com/AWSCloudFormation/latest/UserGuide/aws-resource-athena-workgroup.html)

### LINKED ISSUE #789: AWS::Athena::DataCatalog

### LINKED ISSUE #790: AWS::Athena::NamedQuery

### LINKED ISSUE #791: AWS::Athena::PreparedStatement

### LINKED ISSUE #792: AWS::Athena::WorkGroup

### PR DESCRIPTION
**Pull request by amanenk**
closes #788
and its sub-tasks:
closes #789 
closes #790 
closes #791 
closes #792 
fixed migration

### CODE REVIEW

File: resources/services/athena/data_catalogs.go [hunk 1]
```diff
@@ -0,0 +1,306 @@
...
+				// retrieving of default data catalog (AwsDataCatalog) returns "not found error" but it exists and its
+				// relations can be fetched by its name
+				if *d.CatalogName == "AwsDataCatalog" {
+					res <- types.DataCatalog{Name: d.CatalogName, Type: d.Type}
+					continue
+				}
+				return diag.WrapError(err)
+			}
+			res <- *dc.DataCatalog
+		}
```
Comment (bbernays) [inline resources/services/athena/data_catalogs.go:225]: This should be done in a Go Routine like this:
https://github.com/cloudquery/cq-provider-aws/blob/5cd40bd911b6cd0bbc2c14c4edb8726e76e1b5ad/resources/services/ecs/task_definitions.go#L658-L674

File: resources/provider/migrations/postgres/30_v0.11.5.down.sql [hunk 2]
```diff
@@ -0,0 +1,26 @@
...
+DROP TABLE IF EXISTS aws_shield_attacks;
+
+-- Resource: shield.protections
+DROP TABLE IF EXISTS aws_shield_protections;
+
+-- Resource: shield.protections_groups
+DROP TABLE IF EXISTS aws_shield_protection_groups;
+
+-- Resource: shield.subscriptions
+DROP TABLE IF EXISTS aws_shield_subscriptions;
```
[the remark this answers is not in the document]
Comment (amanenk) [reply resources/provider/migrations/postgres/30_v0.11.5.down.sql:13]: they where accidentally pushed under wrong version. I have fixed them there. should I fix them in separate pr?

[the remark this answers is not in the document]
Comment (bbernays) [reply resources/provider/migrations/postgres/30_v0.11.5.down.sql:13]: @disq - Can you advise on how to proceed?

[the remark this answers is not in the document]
Comment (disq) [reply resources/provider/migrations/postgres/30_v0.11.5.down.sql:13]: @bbernays it's all OK to change things until a release is cut. if released then it's out in the wild and should live with it, potentially doing the fixes in new versions (so no edits to migration files after it gets tagged)

**Review by bbernays — APPROVED** [review]
LGTM

File: resources/provider/provider.go [hunk 3]
```diff
@@ -3,6 +3,8 @@ package provider
 import (
 	"embed"
 
+	"github.com/cloudquery/cq-provider-aws/resources/services/athena"
+
```
Comment (disq) [inline resources/provider/provider.go:7]: another rogue line, line 6 should be in the other block

**Review by disq — APPROVED** [review]
LGTM with nits

**Review by roneli — APPROVED** [review]
LGTM

**Review by disq — APPROVED** [review]

**Pull request merged**